storage/scylla_api.h

```diff
@@ -0,0 +1,188 @@
+  "OO"
+#define TRANSACTION_7                                                               \
+  "TRANSACTION700000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_8                                                               \
+  "TRANSACTION800000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+#define TRANSACTION_9                                                               \
+  "TRANSACTION900000LIKCEJTTIQOTTAWSQCCQQ9A9VOKIWRBWVPXMCGUENWVVMQAMPEIVHEQ9JXLCNZ" \
+  "OO"
+
+#define TIMESTAMP_1 1
+#define TIMESTAMP_2 2
+#define TIMESTAMP_3 3
+
+#define MESSAGE_1 "MESSAGE1"
+#define MESSAGE_2 "MESSAGE2"
+#define MESSAGE_3 "MESSAGE3"
+#define MESSAGE_4 "MESSAGE4"
+#define MESSAGE_5 "MESSAGE5"
+#define MESSAGE_6 "MESSAGE6"
+
+typedef struct Transaction_ {
+  cass_byte_t* bundle;
+  cass_byte_t* address;
+  cass_byte_t* transaction;
+  char* message;
+  cass_int64_t value;
+  cass_int64_t timestamp;
+  cass_byte_t* trunk;
+  cass_byte_t* branch;
+} Transaction;
+
+Transaction* new_Transaction(void);
+void free_Transaction(Transaction** obj);
+
+/**
+ * Initialize logger
+ */
+void scylla_api_logger_init();
+
+/**
+ * Release logger
+ *
+ * @return
+ * - zero on success
+ * - EXIT_FAILURE on error
+ */
+int scylla_api_logger_release();
+
+/**
+ * @brief connect to Scylla node and create table
+ *
+ * @param[out] cluster Scylla node cluster
+ * @param[out] session Scylla cluster session
+ * @param[in] hosts Scylla node ip
+ * @param[in] is_need_create_table true : create, false : not to create
+ *
+ * @return
+ * - SC_OK on success
+ * - non-zero on error
+ */
+status_t init_scylla(CassCluster** cluster, CassSession* session, char* hosts, bool is_need_create_table);
+
+/**
+ * @brief insert transactions into bundle table
+ *
+ * @param[in] session Scylla cluster session
+ * @param[in] transaction input transactions data
+ * @param[in] tran_num input transactions number
+ *
+ * @return
+ * - SC_OK on success
+ * - non-zero on error
+ */
+status_t insert_transaction_into_bundleTable(CassSession* session, Transaction* transaction, size_t trans_num);
+
+/**
+ * @brief insert transactions into edge table
+ *
+ * @param[in] session Scylla cluster session
+ * @param[in] transaction input transactions data
+ * @param[in] tran_num input transactions number
+ *
+ * @return
+ * - SC_OK on success
+ * - non-zero on error
+ */
+status_t insert_transaction_into_edgeTable(CassSession* session, Transaction* transaction, size_t trans_num);
+
+/**
+ * @brief get bundles from edge table by edge
+ *
+ * @param[in] session Scylla cluster session
+ * @param[in] edge primary key for select bundles
+ * @param[in] column_name the column we want to select
+ * @param[out] res_queue response transactions queue
+ *
+ * @return
+ * - SC_OK on success
+ * - non-zero on error
+ */
+status_t get_column_from_edgeTable(CassSession* session, hash243_queue_t* res_queue, cass_byte_t* edge,
```

Comment:
I just wonder if it makes sense to introduce new `foreach` macro for transactions.

Reply:
@locke2833 like this one

Reply:
If we need to iterate a collection of transaction which had been formed as a queue, new `foreach` macro will make iterate much easily. But currently, in our design, we would only supply APIs both getTrytes and findTransactions which only need a queue of hashes.  
The IF chronicle project could only supply those two APIs now either.